modules/tensor_mechanics/src/materials/ComputeAxisymmetric1DFiniteStrain.C

```diff
@@ -27,21 +26,34 @@ ComputeAxisymmetric1DFiniteStrain::ComputeAxisymmetric1DFiniteStrain(
     const InputParameters & parameters)
   : Compute1DFiniteStrain(parameters),
     _disp_old_0(coupledValueOld("displacements", 0)),
-    _out_of_plane_strain_coupled(isCoupled("out_of_plane_strain")),
-    _out_of_plane_strain(_out_of_plane_strain_coupled ? coupledValue("out_of_plane_strain")
+    _scalar_var_id_provider(
+        isParamValid("scalar_variable_index_provider")
+            ? &getUserObject<ScalarVariableIndexProvider>("scalar_variable_index_provider")
+            : nullptr),
+    _has_out_of_plane_strain(isParamValid("out_of_plane_strain")),
+    _out_of_plane_strain(_has_out_of_plane_strain ? coupledValue("out_of_plane_strain") : _zero),
+    _out_of_plane_strain_old(_has_out_of_plane_strain ? coupledValueOld("out_of_plane_strain")
                                                       : _zero),
-    _out_of_plane_strain_old(_out_of_plane_strain_coupled ? coupledValueOld("out_of_plane_strain")
-                                                          : _zero),
-    _scalar_out_of_plane_strain_coupled(isCoupledScalar("scalar_out_of_plane_strain")),
-    _scalar_out_of_plane_strain(_scalar_out_of_plane_strain_coupled
-                                    ? coupledScalarValue("scalar_out_of_plane_strain")
-                                    : _zero),
-    _scalar_out_of_plane_strain_old(_scalar_out_of_plane_strain_coupled
-                                        ? coupledScalarValueOld("scalar_out_of_plane_strain")
-                                        : _zero)
+    _has_scalar_out_of_plane_strain(isParamValid("scalar_out_of_plane_strain")),
+    _nscalar_strains(
+        _has_scalar_out_of_plane_strain ? coupledScalarComponents("scalar_out_of_plane_strain") : 0)
 {
-  if (_out_of_plane_strain_coupled && _scalar_out_of_plane_strain_coupled)
+  if (_has_out_of_plane_strain && _has_scalar_out_of_plane_strain)
     mooseError("Must define only one of out_of_plane_strain or scalar_out_of_plane_strain");
+
+  if (!_has_out_of_plane_strain && !_has_scalar_out_of_plane_strain)
+    mooseError("Must define either out_of_plane_strain or scalar_out_of_plane_strain");
```

Comment:
This is what is making those BISON tests fail.  The default behavior was that if the out of plane strain wasn't provided, it would be assumed to be zero.  Shouldn't that still be the case now?